File: app/src/main/java/com/semisonfire/cloudgallery/ui/main/disk/adapter/items/DiskItem.java
package com.semisonfire.cloudgallery.ui.main.disk.adapter.items;

public abstract class DiskItem {
    public static final int TYPE_HEADER = 0;
    public static final int TYPE_GALLERY = 1;
    public static final int TYPE_UPLOAD = 2;

    /** Item type. */
    public abstract int getType();

    public abstract String getDate();
}

File: app/src/main/java/com/semisonfire/cloudgallery/ui/disk/adapter/items/DiskItem.kt
package com.semisonfire.cloudgallery.ui.disk.adapter.items

const val TYPE_HEADER = 0
const val TYPE_GALLERY = 1
const val TYPE_UPLOAD = 2
const val TYPE_PROGRESS = 3

abstract class DiskItem {
    /** Item type.  */
    abstract val type: Int
}

class ProgressItem : DiskItem() {
    override val type: Int
        get() = TYPE_PROGRESS
}